src/app/data/loaders/cdfLoader.ts

```diff
@@ -0,0 +1,119 @@
+import { v4 } from '@lukeed/uuid';
+import type { FileCollection } from 'filelist-utils';
+import type { Value } from 'cheminfo-types';
+import { NetCDFReader } from 'netcdfjs';
+import { xMaxValue } from 'ml-spectra-processing';
+
+import {
+  MeasurementKind,
+  Loader,
+  Measurements,
+  getEmptyMeasurements,
+} from '../DataState';
+
+export const cdfLoader: Loader = async function cdfLoader(
+  fileCollection: FileCollection,
+) {
+  const newMeasurements: Measurements = getEmptyMeasurements();
```

Comment:
Please add a comment with the link to the github code from which you was inspired.